test/refda/test_eval.c

```diff
@@ -308,6 +308,70 @@ TEST_CASE("8211838201F48201F48401012568626F6F6C2D786F72", "8201F4")
 TEST_CASE("8211838201F48206008401012568626F6F6C2D786F72", "8201F4")
 // ari:/AC/(/BOOL/true,/TEXTSTR/test,//1/1/OPER/bool-xor) -> /BOOL/false
 TEST_CASE("8211838201F5820A64746573748401012568626F6F6C2D786F72", "8201F4")
+
```

Comment:
Just as pickiness, it would be good to verify that `BYTE` type literal gets handled properly, at least for one of the inputs with a binary operator. I think it will work fine but good to assert in a test case.

Reply:
That's a good call. So, the initial test case failed because `cace_ari_equal` does not handle the case where the compared ARI's have a different primitive type (BYTE is int from CBOR decoding but uint from eval).

Anyway, this was solved by adding numeric promotion to `cace_ari_equal` which also gives us generalized eq/ne operators in the ADM.